### PR TITLE
Fix webscoket TLS bug

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation

Fix webscoket tls bug and add test case.

### Modifications

1. Add mapping : serviceConfiguration.setBrokerClientTlsEnabled(config.isTlsEnabledWithBroker());.

2. Add ProxyServiceTlsStarterTest.java

3. Ajuest ProxyServiceStarter to be more easy to test.

### Verifying this change

- [X] Make sure that the change passes the CI checks.

### CODE REVIEW

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java [hunk 1]
```diff
@@ -412,6 +412,14 @@
     private String tlsTrustStorePassword = null;
 
     /**** --- KeyStore TLS config variables used for proxy to auth with broker--- ****/
+    @FieldContext(
+            category = CATEGORY_KEYSTORE_TLS,
+            dynamic = true,
+            doc = "Enable TLS when talking with other brokers in the same cluster (admin operation)"
+                    + " or different clusters (replication)"
```
Comment (codelipenghui) [inline pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java:419]: Looks the doc is from the broker.conf? We need a more suitable description

**Review by codelipenghui — APPROVED** [review]
LGTM, just left a minor comment about the description

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java:419]: ok.

**Review by gaoran10 — APPROVED** [review]
LGTM

**Review by congbobo184 — APPROVED** [review]

**Pull request merged**